assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -6530,6 +6530,7 @@ private void setNewAssignmentParameters(RunData data, boolean validify) {
         ParameterParser params = data.getParameters();
 
         String assignmentRef = params.getString("assignmentId");
+        Reference assignmentReference = entityManager.newReference(assignmentRef);
```

Comment:
More efficient to use:
AssignmentReferenceReckoner.reckoner().reference(assignmentRef).reckon().getId()
vs
entityManager.newReference(assignmentRef)